src/main/java/com/google/devtools/build/lib/bazel/repository/skylark/SkylarkRepositoryFunction.java

```diff
@@ -232,7 +232,7 @@ public void setUseNativePatch(boolean useNativePatch) {
           new IOException(rule + " must create a directory"), Transience.TRANSIENT);
     }
 
-    if (!outputDirectory.getRelative(LabelConstants.WORKSPACE_FILE_NAME).exists()) {
+    if (!WorkspaceFileHelper.doesWorkspaceFileExistUnder(outputDirectory)) {
```

Comment:
I'm concerned about the correctness of calling Path.exists() (without requesting the FileValue) and of creating files (making a cached FileValue stale without telling Skyframe about it).
Let me ask around how to do this.